proguard/src/main/java/com/flurry/proguard/UploadMapping.java

```diff
@@ -365,22 +377,26 @@ private static void waitForUploadToBeProcessed(String projectId, String uploadId
      * @param token the Flurry auth token
      * @return the upload
      */
-    private static JSONObject fetchUpload(String projectId, String uploadId, String token) {
+    private static JSONObject fetchUpload(String projectId, String uploadId, String token) throws IOException {
         String queryUrl = String.format("%s/project/%s/uploads/%s?fields[upload]=uploadStatus,failureReason",
                     METADATA_BASE, projectId, uploadId);
         HttpGet getRequest = new HttpGet(queryUrl);
         List<Header> requestHeaders = getMetadataHeaders(token);
-        HttpResponse response = executeHttpRequest(getRequest, requestHeaders);
-        expectStatus(response, HttpURLConnection.HTTP_OK);
-        return getJsonFromEntity(response.getEntity());
+        try (CloseableHttpResponse response = executeHttpRequest(getRequest, requestHeaders)) {
+            expectStatus(response, HttpURLConnection.HTTP_OK);
+            return getJsonFromEntity(response.getEntity());
+        } finally {
+            getRequest.releaseConnection();
+        }
     }
 
-    private static HttpResponse executeHttpRequest(HttpUriRequest request, List<Header> requestHeaders) {
+    private static CloseableHttpResponse executeHttpRequest(HttpUriRequest request, List<Header> requestHeaders) {
         for (Header header : requestHeaders) {
             request.setHeader(header.getName(), header.getValue());
         }
         try {
-            return HTTP_CLIENT.execute(request);
+            httpClient = HttpClientBuilder.create().setDefaultRequestConfig(REQUEST_CONFIG).build();
```

Comment:
I think we'll build too many of these if we do this. One call to `uploadFiles` will create at least 3 different `httpClient`s. Let's move this construction to the beginning of `uploadFiles`. Is this client `AutoClosable`, can we wrap `uploadFiles`'s body in a try-with-resources block?